apps/aecore/src/aec_tx_pool.erl

```diff
@@ -108,10 +108,22 @@ init([]) ->
 handle_call({get_max_nonce, Sender}, _From, #state{db = Mempool} = State) ->
     {reply, int_get_max_nonce(Mempool, Sender), State};
 handle_call({push, Txs, Event}, _From, #state{db = Mempool} = State) ->
-    [pool_db_put(Mempool, pool_db_key(Tx), Tx, Event) || Tx <- Txs],
+    lists:foreach(
+        fun(Tx) ->
+            pool_db_put(Mempool, pool_db_key(Tx), Tx, Event),
+            TxHash = aetx:hash(aetx_sign:tx(Tx)),
+            aec_db:write_tx(TxHash, mempool, Tx)
```

Comment:
So tx is stored in ETS *and *Mnesia?

Reply:
Yes, mempool keeps its state in ets and we also persist the transactions in mnesia (with some indexing applied)